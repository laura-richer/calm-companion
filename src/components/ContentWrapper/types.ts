export type ContentWrapperProps = {
  children: string | JSX.Element | JSX.Element[];
};
