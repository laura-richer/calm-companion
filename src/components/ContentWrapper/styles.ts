import { StyleSheet } from 'react-native';
import { spacer } from '~styles/variables';

export default StyleSheet.create({
  contentWrapper: {
    paddingHorizontal: spacer.XS,
    paddingVertical: spacer.MD,
  },
  contentWrapperContainer: {
    paddingBottom: 210,
  },
});
