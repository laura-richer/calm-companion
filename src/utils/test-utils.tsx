import React, { PropsWithChildren } from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { render, RenderOptions } from '@testing-library/react-native';
import { Provider as StoreProvider } from 'react-redux';
import { EnhancedStore, configureStore } from '@reduxjs/toolkit';
import { SafeAreaProvider } from 'react-native-safe-area-context';
import { RootState } from '~store/index';
import navigationReducer, {
  initialState as navigationInitialState,
} from '~store/slices/navigation';
import quickHelpReducer, { initialState as quickHelpInitialState } from '~store/slices/quickHelp';

type DeepPartial<T> = {
  [K in keyof T]?: T[K] extends object ? DeepPartial<T[K]> : T[K];
};

interface ExtendedRenderOptions extends Omit<RenderOptions, 'queries'> {
  preloadedState?: Partial<RootState>;
  store?: EnhancedStore;
}

export const renderWithProviders = (
  ui: React.ReactElement,
  { preloadedState = {}, store = createTestStore(preloadedState) }: ExtendedRenderOptions = {}
) => {
  const wrapper = ({ children }: PropsWithChildren<{}>): JSX.Element => {
    return (
      <StoreProvider store={store}>
        <SafeAreaProvider>
          <NavigationContainer>{children}</NavigationContainer>
        </SafeAreaProvider>
      </StoreProvider>
    );
  };

  return {
    ...render(ui, {
      wrapper,
    }),
  };
};

const _createStore = (preloadedState: Partial<RootState> = {}): EnhancedStore => {
  return configureStore({
    reducer: {
      navigation: navigationReducer,
      quickHelp: quickHelpReducer,
    },
    preloadedState,
    middleware: getDefaultMiddleware =>
      getDefaultMiddleware({
        immutableCheck: { warnAfter: 256 },
        serializableCheck: { warnAfter: 256 },
      }),
  });
};

export const createTestStore = (preloadedState: DeepPartial<RootState> = {}): EnhancedStore => {
  const state: RootState = {
    // @ts-ignore
    navigation: {
      ...navigationInitialState,
      ...(preloadedState.navigation || {}),
    },
    // @ts-ignore
    quickHelp: {
      ...quickHelpInitialState,
      ...(preloadedState.quickHelp || {}),
    },
  };
  const store: EnhancedStore = _createStore(state);
  const origDispatch = store.dispatch;
  store.dispatch = jest.fn(origDispatch);
  return store;
};
