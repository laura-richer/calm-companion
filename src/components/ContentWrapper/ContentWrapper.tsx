import React from 'react';
import { ScrollView, View } from 'react-native';
import { ContentWrapperProps } from './types';
import styles from './styles';

const ContentWrapper = ({ children }: ContentWrapperProps) => {
  return (
    <ScrollView>
      <View style={styles.contentWrapperContainer}>{children}</View>
    </ScrollView>
  );
};

export default ContentWrapper;
