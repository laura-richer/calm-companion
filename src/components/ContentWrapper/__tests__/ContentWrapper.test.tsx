import React from 'react';
import { Text } from 'react-native';
import { screen } from '@testing-library/react-native';
import { renderWithProviders } from '~utils/test-utils';
import ContentWrapper from '../ContentWrapper';

const _renderDefault = () => {
  return renderWithProviders(
    <ContentWrapper>
      <Text>Child content</Text>
    </ContentWrapper>
  );
};

describe('content wrapper', () => {
  beforeEach(() => {
    _renderDefault();
  });

  test('matches snapshot', () => {
    expect(screen.toJSON()).toMatchSnapshot();
  });

  test('renders child content', () => {
    const content = screen.getByText('Child content');
    expect(content).toBeOnTheScreen();
  });
});
