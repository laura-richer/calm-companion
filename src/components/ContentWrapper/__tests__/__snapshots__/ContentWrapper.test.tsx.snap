// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`content wrapper matches snapshot 1`] = `
<RCTScrollView
  style={
    {
      "paddingHorizontal": 10,
      "paddingVertical": 30,
    }
  }
>
  <View>
    <View
      style={
        {
          "paddingBottom": 210,
        }
      }
    >
      <Text>
        Child content
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
